### PR TITLE
SPCTR-2763 Fix: Forms - Issue on submit failure in FSE theme.

### PR DESCRIPTION
**Pull request by AbhijitNage123**
### Description

- https://brainstormforce.atlassian.net/browse/SPCTR-2763

### Screenshots

- https://share.getcloudapp.com/nOuKrXrn

### Types of changes

- Bug fix (non-breaking change which fixes an issue)

### How has this been tested?

- Added to the card.

### Checklist:
- [x] My code is tested
- [ ] My code follows accessibility standards 
- [ ] My code has proper inline documentation 
- [x] I've included any necessary tests 
- [ ] I've included developer documentation 
- [x] I've added proper labels to this pull request

### CODE REVIEW

**Review by rajmohanreddy774 — APPROVED** [review]

**Review by RajkiranBagal — APPROVED** [review]

File: blocks-config/forms/class-uagb-forms.php [hunk 1]
```diff
@@ -120,15 +120,30 @@ public function process_forms() {
...
-			$blocks                   = parse_blocks( $post_content );
-			$current_block_attributes = false;
-			if ( ! empty( $blocks ) && is_array( $blocks ) ) {
-				$current_block_attributes = $this->recursive_inner_forms( $blocks, $block_id );
+			if ( wp_is_block_theme() ) {
+				$wp_query_args  = array(
+					'post_status' => array( 'publish' ),
+					'post_type'   => 'wp_template',
+				);
+				$template_query = new WP_Query( $wp_query_args );
```
Comment (RajkiranBagal) [inline blocks-config/forms/class-uagb-forms.php:132]: Check !empty  & is_array for `$template_query->posts`

File: blocks-config/forms/class-uagb-forms.php [hunk 2]
```diff
@@ -120,15 +120,30 @@ public function process_forms() {
...
+				foreach ( $template_query->posts as $post ) {
+					$template = _build_block_template_result_from_post( $post );
+					if ( is_wp_error( $template ) ) {
+						continue;
+					}
+					if ( get_template() === $template->theme ) {
+						$current_block_attributes = $this->recursive_inner_forms( parse_blocks( $template->content ), $block_id );
```
Comment (RajkiranBagal) [inline blocks-config/forms/class-uagb-forms.php:139]: Check `! empty( $blocks ) && is_array( $blocks )` here too as you have checked below.

File: blocks-config/forms/class-uagb-forms.php [hunk 3]
```diff
@@ -120,15 +120,30 @@ public function process_forms() {
...
+					'post_status' => array( 'publish' ),
+					'post_type'   => 'wp_template',
+				);
+				$template_query = new WP_Query( $wp_query_args );
+				foreach ( $template_query->posts as $post ) {
+					$template = _build_block_template_result_from_post( $post );
+					if ( is_wp_error( $template ) ) {
+						continue;
+					}
+					if ( get_template() === $template->theme ) {
```
Comment (RajkiranBagal) [inline blocks-config/forms/class-uagb-forms.php:138]: Is this Right to call functions inside if the condition checks?
cc - @adi3890

**Review by RajkiranBagal — CHANGES_REQUESTED** [review]

File: blocks-config/forms/class-uagb-forms.php (hunk 1 above)
Comment (AbhijitNage123) [reply blocks-config/forms/class-uagb-forms.php:138]: Updated as suggested

File: blocks-config/forms/class-uagb-forms.php (hunk 2 above)
Comment (AbhijitNage123) [reply blocks-config/forms/class-uagb-forms.php]: Done

File: blocks-config/forms/class-uagb-forms.php (hunk 3 above)
Comment (AbhijitNage123) [reply blocks-config/forms/class-uagb-forms.php]: @RajkiranBagal Checked `function_exists()` for `_build_block_template_result_from_post`

**Review by RajkiranBagal — APPROVED** [review]

**Review by RajkiranBagal — APPROVED** [review]

**Pull request merged**